### PR TITLE
Handle registers on push on 8051

### PR DESCRIPTION
**Pull request by gogo2464**
**Checklist**

- [ ] Closing issues: #issue
- [x] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [radare2book](https://github.com/radareorg/radare2book)

**Description**

### CODE REVIEW

File: libr/asm/arch/8051/8051_ass.c [hunk 1]
```diff
@@ -545,14 +545,76 @@ static bool singlearg_reladdr(ut8 const firstbyte, char const* arg
...
 	, ut8 **out)
 {
-	ut8 address;
-	if (!address_direct (arg, &address)) {
-		return false;
+	bool ret = true;
+	ut8 address = 0x00;
+	if (!address_direct (arg, &address) && !is_reg (arg)) {
+		ret = false;
+	} else if (is_reg (arg)) {
```
Comment (trufae) [inline libr/asm/arch/8051/8051_ass.c:552]: Move into a separate function and use a steuct snd a for loop to iterate over all the strings . Something like address = getreg(“r0”);

**Comment by gogo2464** [comment]
@trufae ready to merge

File: libr/asm/arch/8051/8051_ass.c [hunk 2]
```diff
@@ -377,19 +377,98 @@ static bool to_address(char const* addr_str, ut16* addr_out) {
...
+	bool found;
 	if ( !parse_hexadecimal (addr_str, &addr_big)
 		|| (0xFF < addr_big)) {
-		return false;
+		found = false;
+	} else {
+		found = true;
+	}
+
+	if (found == false) {
```
Comment (trufae) [inline libr/asm/arch/8051/8051_ass.c:461]:
```suggestion
	if (!found) {
```

**Review by trufae — CHANGES_REQUESTED** [review]

**Comment by trufae** [comment]
Uhm the ifelse cascade is still there. But anyway i think its good to merge and we can improve it later

**Pull request merged**